src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfoExtensions.cs

```diff
@@ -27,13 +28,13 @@ public static CSharpCompilationOptions CreateCompilationOptions(this ProjectFile
                 compilationOptions = compilationOptions.WithGeneralDiagnosticOption(ReportDiagnostic.Error);
             }
 
+            compilationOptions = compilationOptions.WithSpecificDiagnosticOptions(projectFileInfo.GetDiagnosticOptions());
+
```

Comment:
Does order make any difference (why was this moved above `NullableContextOptions`)? If not, consider moving this above all the `if`s for better readability.